### PR TITLE
Windows: Eina: Einafy unPOSIX sys/time.h

### PR DESCRIPTION
**Pull request by Coquinho**
Windows: Eina: Einafy unPOSIX sys/time.h

Summary:
========
This PR reorganizes `unposix/sys/time.h` and as specified by issue #40,
this PR moves `unposix/sys/time.{h,c}` to `eina/eina_time.{h,c}` which
implements `eina_` wrap functions to `gettimeofday` and `localtime_r`.

In case of a Windows build it also defines struct `timezone` and `u_short`;
In case of a Linux build it includes `sys/time.h`;

Every occurrence of `gettimeofday` was moved to `eina_gettimeofday`;
Every occurrence of `localtime_r` was moved to `eina_localtime_r`;
Every `#include` of `sys/time.h` was removed and it was changed to
`#include "eina_time.h` in case of an in `eina` include or
`#include <Eina.h>` in case of an out of `eina` include;

`evil/evil_time.{c,h}` `sys/time.h` include was changed to `time.h` as it
only used `time_t`;
`evil/evil_unistd.c` again only uses `time_t`, and as `it includes
`evil_private.h` it will implicitly include `evil_time.h`;

Linux Test Plan:
================
- configure meson with `bindigns=cxx,mono`;
- `ninja` should return only errors/warnings that `master` does;
- `meson test all` should only pass on tests that master does;

Windows Test Plan:
==================
- run `configure.bat`;
- run `build.bat` with `-k0`;
- there should not be any errors/warnings about any related to `sys/time.h`;
- It should yield the same errors/warnings that `devs/expertise/native-windows`
  does;

### CODE REVIEW

File: src/lib/eina/eina_time.c [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#endif
+
+#include "eina_config.h"
+#include "eina_private.h"
+#include "eina_time.h"
+
+#include <stdint.h>
+#include <time.h>
+#include <sys/types.h>
+#if !defined(_WIN32) && defined(HAVE_SYS_TiME_H)
```
Comment (JPTIZ) [inline src/lib/eina/eina_time.c:31]: Wops! `HAVE_SYS_TIME_H` you mean?

File: src/bin/edje/epp/cpplib.c [hunk 2]
```diff
@@ -76,14 +76,13 @@ const char         *version_string = "0.0.0";
 
 #ifndef VMS
 #ifndef USG
+#include <sys/times.h>
 #include <time.h>
-#include <sys/time.h>		/* for __DATE__ and __TIME__ */
 #ifdef HAVE_SYS_RESOURCE_H
```
[the remark this answers is not in the document]
Comment (Coquinho) [reply src/bin/edje/epp/cpplib.c:81]: Wow....It isn't a reorder. It's actually wrong. I removed `sys/time.h` and add `sys/timeS.h`, thanks for noting.

[the remark this answers is not in the document]
Comment (Coquinho) [reply src/bin/edje/epp/cpplib.c:81]: Oh, latter I removed a `sys/times.h`. hum..I will change it back to its original

**Comment by JPTIZ** [comment]
Why is this deleting `pcre.wrap` and `zlib.wrap`?

**Comment by Coquinho** [comment]
> Why is this deleting `pcre.wrap` and `zlib.wrap`?

I don't know. Now it should be fine. Sorry about that.

**Pull request closed without merging**